dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+      implicit ec: ExecutionContext): Future[Transaction] = {
+    val spks = txBuilder.destinations.toVector.map(_.scriptPubKey)
+
+    subtractFeeFromOutputsAndSign(txBuilder, spks)
+  }
+
+  // This invariant ensures that emptyChangeSPK is never used above
+  val noEmptyOutputs: (Seq[BitcoinUTXOSpendingInfoFull], Transaction) => Boolean = {
+    (_, tx) =>
+      tx.outputs.forall(_.scriptPubKey != EmptyScriptPubKey)
+  }
+
+  /** Subtracts the estimated fee by removing from each output with a specified spk evenly */
+  def subtractFeeFromOutputsAndSign(
+      txBuilder: BitcoinTxBuilder,
+      spks: Vector[ScriptPubKey])(
+      implicit ec: ExecutionContext): Future[Transaction] = {
+    subtractFeeFromOutputs(txBuilder, spks).flatMap(_.sign(noEmptyOutputs))
+  }
+
+  private def subtractFees(
+      tx: Transaction,
+      feeRate: FeeUnit,
+      spks: Vector[ScriptPubKey]): Vector[TransactionOutput] = {
+    val fee = feeRate.calc(tx)
+
+    val outputs = tx.outputs.zipWithIndex.filter {
+      case (output, _) => spks.contains(output.scriptPubKey)
+    }
+    val unchangedOutputs = tx.outputs.zipWithIndex.filterNot {
+      case (output, _) => spks.contains(output.scriptPubKey)
+    }
+
+    val feePerOutput = Satoshis(Int64(fee.satoshis.toLong / outputs.length))
+    val feeRemainder = Satoshis(Int64(fee.satoshis.toLong % outputs.length))
+
+    val newOutputsWithoutRemainder = outputs.map {
+      case (output, index) =>
+        (TransactionOutput(output.value - feePerOutput, output.scriptPubKey),
+         index)
+    }
+    val (lastOutput, lastOutputIndex) = newOutputsWithoutRemainder.last
+    val newLastOutput = TransactionOutput(lastOutput.value - feeRemainder,
+                                          lastOutput.scriptPubKey)
+    val newOutputs = newOutputsWithoutRemainder
+      .dropRight(1)
+      .:+((newLastOutput, lastOutputIndex))
+
+    (newOutputs ++ unchangedOutputs).sortBy(_._2).map(_._1).toVector
+  }
+
+  /** Subtracts the estimated fee by removing from each output with a specified spk evenly */
+  def subtractFeeFromOutputs(
+      txBuilder: BitcoinTxBuilder,
+      spks: Vector[ScriptPubKey])(
+      implicit ec: ExecutionContext): Future[BitcoinTxBuilder] = {
+    txBuilder.unsignedTx.flatMap { tx =>
+      val allOuputsWithNew = subtractFees(tx, txBuilder.feeRate, spks)
+
+      BitcoinTxBuilder(allOuputsWithNew,
+                       txBuilder.utxoMap,
+                       txBuilder.feeRate,
+                       txBuilder.changeSPK,
+                       txBuilder.network,
+                       txBuilder.lockTimeOverrideOpt)
+    }
+  }
+
+  /** Subtracts the estimated fee by removing from each output with a specified spk evenly */
+  def subtractFeeFromOutputs(
+      tx: BaseTransaction,
+      feeRate: FeeUnit,
+      spks: Vector[ScriptPubKey]): BaseTransaction = {
+    // tx has empty script sigs and we need to account for witness data in fees
+    val dummyTxWit = TransactionWitness(
```

Comment:
important change: fee estimation is done by filling a `TransactionWitness` with `DummyECDigitalSignature`s rather than using `BitcoinTxBuilder.loop(dummy = true)`. It seems that this creates exactly the same fees for funding tx but for some reason I don't understand, the fees on the refund tx (spends the funding tx only) have been reduced by a couple satoshis